Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -149,38 +149,63 @@ RegexResult RegExpPrototype::do_match(const Regex<ECMA262>& re, const StringView
     return result;
 }
 
-// 22.2.5.2 RegExp.prototype.exec ( string ), https://tc39.es/ecma262/#sec-regexp.prototype.exec
-JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::exec)
+// 22.2.5.2.1 RegExpExec ( R, S ), https://tc39.es/ecma262/#sec-regexpexec
+Value RegExpPrototype::regexp_exec(GlobalObject& global_object, Object& rx, String const& string)
```

Comment:
What's the reason to put this on the prototype? Should be standalone (possibly static), IMO.

Reply:
Ah, because `regexp_builtin_exec` needs `do_match` which is on the prototype already. I meant to get rid of `do_match` though since it's now only used by `regexp_builtin_exec` (used to be used in a couple places), so these don't need to be on the prototype anymore either.